### ISSUE TITLE
[QUESTION] Dry-run simulation_Fortran runtime error

### ISSUE THREAD

**Issue by jiaying002** [geoschem/geos-chem#1475]
I got an error when I did the dry-run simulation. It looks like this:
![screenshot](https://user-images.githubusercontent.com/113654537/199200867-fda6705e-1827-4e35-9ad7-135d9c4e19bf.png)

For the unit number issue, I am not sure what the meaning of the unit number is negative and how to solve it. For the open file issue, I am pretty sure all the files are opened, and no file is missing. I would super appreciate it if anyone could help me with it.

_In case, here is my log.run file_
[log.dryrun.txt](https://github.com/geoschem/geos-chem/files/9908498/log.dryrun.txt)

**Comment by yantosca** [comment]
Thanks for writing @jiaying002.  It looks like your dry run might have finished OK from the log file:
```console
HEMCO: Opening /home/jiaying002/GCClassic/src/GEOS_2x2.5/MERRA2/2018/01/MERRA2.20180131.A1.2x25.nc4
HEMCO: Opening /home/jiaying002/GCClassic/src/GEOS_2x2.5/MERRA2/2018/01/MERRA2.20180131.A3cld.2x25.nc4
HEMCO: Opening /home/jiaying002/GCClassic/src/GEOS_2x2.5/MERRA2/2018/01/MERRA2.20180131.A3dyn.2x25.nc4
HEMCO: Opening /home/jiaying002/GCClassic/src/GEOS_2x2.5/MERRA2/2018/01/MERRA2.20180131.A3mstC.2x25.nc4
HEMCO: Opening /home/jiaying002/GCClassic/src/GEOS_2x2.5/MERRA2/2018/01/MERRA2.20180131.A3mstE.2x25.nc4
HEMCO: Opening /home/jiaying002/GCClassic/src/GEOS_2x2.5/MERRA2/2018/01/MERRA2.20180131.I3.2x25.nc4
HEMCO: Opening /home/jiaying002/GCClassic/src/GEOS_2x2.5/MERRA2/2018/02/MERRA2.20180201.I3.2x25.nc4
HEMCO: Opening /home/jiaying002/GCClassic/src/HEMCO/Yuan_XLAI/v2021-06/Yuan_proc_MODIS_XLAI.025x025.2018.nc
HEMCO: Opening /home/jiaying002/GCClassic/src/HEMCO/OFFLINE_LIGHTNING/v2020-03/MERRA2/2018/FLASH_CTH_MERRA2_0.5x0.625_2018_01.nc4

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!! GEOS-CHEM IS IN DRY-RUN MODE!
!!!
!!! You will NOT get output for this run!
!!! Use this command to validate a GEOS-Chem run configuration:
!!!   ./gcclassic --dryrun > log
!!!
!!! REMOVE THE --dryrun ARGUMENT FROM THE COMMAND LINE
!!! BEFORE RUNNING A GEOS-Chem PRODUCTION SIMULATION!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!! Start Date       : 20180101 000000
!!! End Date         : 20180131 000000
!!! Simulation       : fullchem
!!! Meteorology      : MERRA2
!!! Grid Resolution  : 2.0x2.5
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```
The error is happening in the subroutine that prints out the dry-run message to the screen and the log file.  This subroutine is at the end of `Interfaces/GCClassic/main.F90`

https://github.com/geoschem/geos-chem/blob/53cac3b937064bdceed21acec721a157b6fb8b1e/Interfaces/GCClassic/main.F90#L2736-L2796

and is called further up in `main.F90` to write the warning to the stdout stream (unit #6) as well as the HEMCO.log file.

https://github.com/geoschem/geos-chem/blob/53cac3b937064bdceed21acec721a157b6fb8b1e/Interfaces/GCClassic/main.F90#L2727-L2731

Which version are you using?  If you are using the 14.0.0 code (pre-release) then we direct the HEMCO.log output to the screen, so it could be possible that the HEMCO.log file unit number `HcoState%Config%Err%LUN` is negative.  We could fix that by making sure it is positive before trying to write the dry-run header to the log file, i.e.:

```F90
    ! For dry-run simulations, Write the dry-run header to
    ! stdout (aka GEOS-Chem log file) and the HEMCO log file.
    IF ( Input_Opt%DryRun ) THEN
       CALL Print_Dry_Run_Warning( 6 )
       IF ( HcoState%Config%Err%Lun > 0 ) THEN
          CALL Print_Dry_Run_Warning( HcoState%Config%Err%LUN )
       ENDIF
    ENDIF
```
and that should prevent the error you saw.

Long story short: your dry-run should be OK but the error is just happening when printing to the log file at the end of the dry-run.

Also tagging @Jourdan-He and @SaptSinha.

**Comment by jiaying002** [comment]
Thank you sooooo much! You saved my day!

**Comment by yangbyy** [comment]
I had the same problem. I made changes to the main.F90 file based on your suggestions but still have the same problem.
![image](https://user-images.githubusercontent.com/115625689/199458277-4461c915-01a1-4662-ac7d-3ccd2111f21a.png)
There are my  main.F90 and log file, please let me know what's wrong, looking forward to getting your reply. 
![image](https://user-images.githubusercontent.com/115625689/199460952-52c24ef4-e9b7-44ee-a004-2bf9db158b9e.png)
[main.txt](https://github.com/geoschem/geos-chem/files/9919302/main.txt)
[log.txt](https://github.com/geoschem/geos-chem/files/9919323/log.txt)

**Comment by yantosca** [comment]
@jiaying002: Did your recompile the code?  When I made this fix this error went away.

**Comment by jiaying002** [comment]
Honestly, I tried the code but still pumped up the same error. My version is 13.9.0, so I thought it might be the reason why the code could not work. Since you said it was just a print-in error, I ignored it and moved on.
